### PR TITLE
Annotations getters setters

### PR DESCRIPTION
**Pull request by gerhardj**
related to #125

### CODE REVIEW

File: openapi-maven-plugin/src/main/java/io/github/kbuntrock/yaml/model/Schema.java [hunk 1]
```diff
@@ -160,13 +163,17 @@ public Schema(final DataObject wrappedDataObject, final boolean mainReference, f
...
+                            List<Annotation> annotations = getRelevantAnnotationsForField(field);
+
 							if(field.isAnnotationPresent(JsonIgnore.class)) {
 								// Field is tagged ignore. No need to document it.
 								continue;
 							}
 							// Jackson @JsonProperty annotation handling
 							String propertyFieldName = field.getName();
-							final JsonProperty jsonPropertyField = field.getAnnotation(JsonProperty.class);
+//							final JsonProperty jsonPropertyField = field.getAnnotation(JsonProperty.class);
```
Comment (kbuntrock) [inline openapi-maven-plugin/src/main/java/io/github/kbuntrock/yaml/model/Schema.java]: Commented line to remove

File: openapi-maven-plugin/src/main/java/io/github/kbuntrock/yaml/model/Schema.java [hunk 2]
```diff
@@ -273,18 +280,43 @@ public Schema(final DataObject wrappedDataObject, final boolean mainReference, f
...
-		final Size size = field.getAnnotation(Size.class);
+    private <T extends Annotation> T findAnnotationByClass(List<Annotation> annotations, Class<T> annotationType) {
+        return (T) annotations.stream().filter(a -> a.annotationType().equals(annotationType)).findFirst().orElse(null);
+    }
+
+    private List<Annotation> getRelevantAnnotationsForField(Field field) {
+        Class<?> declaringClass = field.getDeclaringClass();
+        List<Method> relatedMethods = Arrays.stream(declaringClass.getMethods())
+                .filter(method -> (
+                                          method.getName().equalsIgnoreCase(field.getName())
```
Comment (kbuntrock) [inline openapi-maven-plugin/src/main/java/io/github/kbuntrock/yaml/model/Schema.java]: Could you split lambda with two dedicated functions? 
- One for getters (get + is), checking the return type (already done) + the fact that the parameter count should be equal to zero (not done).
- One for setters, with the parameter count equal to one (already done), type equality (already done) and return type equal to void (not done)

File: openapi-maven-plugin/src/main/java/io/github/kbuntrock/yaml/model/Schema.java [hunk 3]
```diff
@@ -273,18 +280,43 @@ public Schema(final DataObject wrappedDataObject, final boolean mainReference, f
...
+                                  )
+                )
+                .collect(Collectors.toList());
+        return Stream.concat(relatedMethods.stream().flatMap(method -> Arrays.stream(method.getAnnotations())),
+                        Arrays.stream(field.getAnnotations()))
+                .collect(Collectors.toList());
+    }
+
+    private void extractConstraints(final Field field, List<Annotation> annotations, final Property property) {
+//		final Size size = field.getAnnotation(Size.class);
```
Comment (kbuntrock) [inline openapi-maven-plugin/src/main/java/io/github/kbuntrock/yaml/model/Schema.java]: Commented line to remove

**Review by kbuntrock — CHANGES_REQUESTED** [review]
Looks pretty nice. Just requested a few minor ajustements.

So sorry for the long delay. Sometimes time just flies by too fast!

Once it's merged I'll do the next release shortly after.

**Comment by gerhardj-b** [comment]
Many thanks for your review. The requested changes should be done now.

**Review by kbuntrock — APPROVED** [review]
Nice, thank you for your contribution ❤

**Pull request merged**